### DISCUSSION TITLE
Allow baggage attributes to be available at the root span without Manual Instrusion

### DISCUSSION THREAD

**Discussion by MadhusudanN** [open-telemetry/opentelemetry-java#5834] [Q&A]
Currently we notice in order to propagate the custom attributes added to baggage we need to add custom span processor as detailed in the below discussions ::

https://github.com/open-telemetry/opentelemetry-java/discussions/3113 – How to use baggage
https://github.com/open-telemetry/opentelemetry-java/discussions/4801 – Java code to add baggage as span attributes to traces

Can someone please let me know if there is a way to get the custom attributes at the root level where baggage is initialized with custom attributes. 
To explain with an example, lets say we have an endpoint and we have Auth filter. We will initialize the baggage with custom attributes at the Auth Filter, so when we hit the API, the request goes through the filter and then reaches the endpoint. Since the baggage is initialized at the Filter, in its span can we have the baggage attributes to be shown.

Water fall diagram below for the above scenario ::

![image-2023-9-15_13-57-36](https://github.com/open-telemetry/opentelemetry-java/assets/26345035/d07adecc-107f-4fb5-bda1-887017c371fd)

In the above diagram , at the filter level if we can get the baggage attributes, that should help.
Please let me know if there is a way to achieve this with Auto instrumentation.

**Comment by jack-berg** [comment]
The typical workflow is to extract context from a carrier, and start a span from that context. If the span is started after baggage has been extracted, then baggage should be in a `SpanProcessor` to extract. 

The code to extract context and start a span would look something like:
```
    OpenTelemetry openTelemetry = // .. get OpenTelemetry instance
    Map<String, String> headers = // .. get headers
    Context context = openTelemetry.getPropagators().getTextMapPropagator()
        .extract(Context.current(), headers,
            new TextMapGetter<Map<String, String>>() {
              @Override
              public Iterable<String> keys(Map<String, String> carrier) {
                return carrier.keySet();
              }

              @Nullable
              @Override
              public String get(@Nullable Map<String, String> carrier, String key) {
                if (carrier == null) {
                  return null;
                }
                for (Map.Entry<String, String> entry : carrier.entrySet()) {
                  if (key.equalsIgnoreCase(entry.getKey())) {
                    return entry.getValue();
                  }
                }
                return null;
              }
            });
    Tracer tracer = openTelemetry.getTracer("my-tracer");
    Span span = tracer.spanBuilder("my-span").startSpan();
    try (Scope scope = span.makeCurrent()) {
      // Do work
      span.end();
    }
```
On the SDK side, you need to make sure `OpenTelemetrySdk` is configured with `W3CBaggagePropagator`, and that you have a span  processor to extract the relevant keys from baggage.

**Comment by MadhusudanN** [comment]
Thanks @jack-berg for the quick help.
